extensions/crc/src/daemon-commander.ts

```diff
@@ -0,0 +1,149 @@
+const got = require('got');
+const os = require('os');
+
+export interface Status {
+  readonly CrcStatus: string;
+  readonly Preset?: string;
+  readonly OpenshiftStatus?: string;
+  readonly OpenshiftVersion?: string;
+  readonly PodmanVersion?: string;
+  readonly DiskUse?: number;
+  readonly DiskSize?: number;
+}
+
+export class DaemonCommander {
+  private apiPath: string;
+
+  constructor() {
+    this.apiPath = `http://unix:${process.env.HOME}/.crc/crc-http.sock:/api`;
+
+    if (os.platform() === 'win32') this.apiPath = 'http://unix://?/pipe/crc-http:/api';
+  }
+
+  async status(): Promise<Status> {
+    const options = {
+      url: this.apiPath + '/status',
+      method: 'GET',
+    };
+
+    const { body } = await got(options);
+    return JSON.parse(body);
+  }
+
+  async logs() {
+    const options = {
+      url: this.apiPath + '/logs',
+      method: 'GET',
+    };
+
+    const { body } = await got(options);
```

Comment:
I don't see where this library is added in dependencies ?

Reply:
`got` is the REST library used by `crc`. Perhaps the one used by `dockerode` could be used here.
The reason we use `got` has to do with named pipe usage on Windows.

Reply:
here I was just saying that the library is used but the package was not added to the list of dependencies

for named pipe or socket you could even use pretty dumb code like
```
const net = require('net');
const conn = net.createConnection('\\\\.\\pipe\\pipe_name’);
```